src/binary-search-tree.js

```diff
@@ -1,10 +1,68 @@
-// https://msdn.microsoft.com/en-us/library/aa289150(v=vs.71).aspx
+
+class Queue {
+  constructor() {
+    this.storage = [];
+  }
+  enqueue(value) {
+    this.storage.unshift(value);
+  }
+  dequeue() {
+    return this.storage.pop();
+  }
+  get size() {
+    return this.storage.length;
+  }
+}
 class BinarySearchTree {
   constructor(value) {
     this.value = value;
     this.left = null;
     this.right = null;
   }
+  insert(data) {
+    if (data > this.value && this.right === null) this.right = new BinarySearchTree(data);
+    else if (data < this.value && this.left === null) this.left = new BinarySearchTree(data); else if (data > this.value) {
+      this.right.insert(data);
+    } else {
+      this.left.insert(data);
+    }
+  }
+  contains(x) {
+    // console.log(this);
+    let flag = false;
+    if (x === this.value) flag = true;
```

Comment:
You should have your code return here after setting the flag to ensure that it stops executing right after it finds the target value.